backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -409,6 +414,82 @@ public FutureExperimentRunDAO(
         .thenCompose(unused -> deleteExperimentRuns(runIds), executor);
   }
 
+  /**
+   * @param newDatasets : new datasets for privilege check
+   * @param errorOut : Throw error while creation (true) otherwise we will keep it silent (false)
+   * @return {@link List} : accessible datasets
+   */
+  private InternalFuture<List<Artifact>> filterAndGetPrivilegedDatasetsOnly(
+      List<Artifact> newDatasets, boolean errorOut) {
+    List<String> newDatasetLinkedIds = new ArrayList<>();
+    for (Artifact dataset : newDatasets) {
+      String datasetVersionId = dataset.getLinkedArtifactId();
+      if (!datasetVersionId.isEmpty()) {
+        newDatasetLinkedIds.add(datasetVersionId);
+      }
+    }
+
+    if (newDatasetLinkedIds.isEmpty()) {
+      return InternalFuture.completedInternalFuture(newDatasets);
+    }
+    return jdbi.withHandle(
+            handle -> {
+              Map<String, Long> datasetVersionDatasetMap = new HashMap<>();
+              handle
+                  .createQuery(
+                      " SELECT commit_hash, repository_id FROM repository_commit WHERE commit_hash IN (<datasetIds>) ")
+                  .bindList("datasetIds", newDatasetLinkedIds)
+                  .map(
+                      (rs, ctx) ->
+                          datasetVersionDatasetMap.put(
+                              rs.getString("commit_hash"), rs.getLong("repository_id")))
```

Comment:
Avoid `map` for side effects as that's bad practice. Instead, return a list of entries and then convert to a map.

Reply:
Remove map and used list as per suggestions